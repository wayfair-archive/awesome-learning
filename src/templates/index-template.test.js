import React from 'react';
import renderer from 'react-test-renderer';
import IndexTemplate from './index-template';

describe('IndexTemplate', () => {
  const props = {
    data: {
      allMarkdownRemark: {
        edges: [
          {
            node: {
              fields: {
                slug: '/test_0',
                categorySlug: '/test_0'
              },
              frontmatter: {
                date: '2016-09-01',
                description: 'test_0',
                category: 'test_0',
                title: 'test_0'
              }
            }
          },
          {
            node: {
              fields: {
                slug: '/test_1',
                categorySlug: '/test_1'
              },
              frontmatter: {
                date: '2016-09-01',
                description: 'test_1',
                category: 'test_1',
                title: 'test_1'
              }
            }
          }
        ]
      },
      site: {
        siteMetadata: {
          title: 'test',
          subtitle: 'test'
        }
      }
    },
    pageContext: {
      currentPage: 1,
      prevPagePath: '/page/1',
      nextPagePath: '/page/3',
      hasNextPage: true,
      hasPrevPage: true
    }
  };

  it('renders correctly', () => {
    const tree = renderer.create(<IndexTemplate {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
