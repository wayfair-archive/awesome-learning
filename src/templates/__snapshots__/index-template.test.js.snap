// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexTemplate renders correctly 1`] = `
Array [
  <StaticQuery
    render={[Function]}
  />,
  <div
    className="Layout Layout--fullBleed"
  >
    <div
      className="Landing"
    >
      <div
        className="Landing-heroWrapper"
      >
        <div
          className="Landing-titleWrapper"
        >
          <div
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            <h2
              className="SectionTitle"
            >
               
              Learn JavaScript and Front-End Fundamentals.
               
            </h2>
          </div>
          <p
            className="Text Text--xl"
          >
            At your own pace.
          </p>
        </div>
        <div
          className="Landing-courseWrapper"
        >
          <Link
            className="TrackCard"
            to="/courses/Array-Methods/"
          >
            <div
              className="TrackCard-iconWrap"
            >
              <svg
                className="Icon TrackCard-icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 24h7.67v-2H1.83V2h5.92V0H0M24 0h-7.67v2h5.8v20h-5.92v2H24"
                />
              </svg>
            </div>
            <div
              data-enzyme-id="Block-RenderedElement"
              style={
                Object {
                  "marginBottom": "8px",
                  "marginLeft": "0px",
                  "marginRight": "0px",
                  "marginTop": "0px",
                }
              }
            >
              <h3
                className="SectionTitle SectionTitle--uppercase"
              >
                 
                Array Methods
                 
              </h3>
            </div>
            <p
              className="Text"
            >
              Learn functional array methods like filter, map, and reduce!
            </p>
          </Link>
          <Link
            className="TrackCard"
            to="/courses/Data-Types/"
          >
            <div
              className="TrackCard-iconWrap"
            >
              <svg
                className="Icon TrackCard-icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22.62 11.76H19.7l1.07-3.57a2.41 2.41 0 0 0 .08-.6 1.57 1.57 0 0 0-1.37-1.71h-2.14l1.07-3.57a2.35 2.35 0 0 0 .08-.6A1.57 1.57 0 0 0 17.12 0H4.55a1.43 1.43 0 0 0-1.23 1L.12 9.83a2.06 2.06 0 0 0-.12.7 1.93 1.93 0 0 0 .4 1.21 1.23 1.23 0 0 0 1 .5h2.34l-1.26 3.47a2 2 0 0 0-.12.69 2 2 0 0 0 .4 1.22 1.26 1.26 0 0 0 1 .5h3.12l-1.25 3.46a2.1 2.1 0 0 0-.12.7 1.91 1.91 0 0 0 .4 1.21 1.23 1.23 0 0 0 1 .51H20a1.44 1.44 0 0 0 1.29-1.11l2.64-8.82a2 2 0 0 0 .09-.61 1.57 1.57 0 0 0-1.4-1.7zm-21.24-1a.18.18 0 0 1-.14-.08.25.25 0 0 1-.06-.17.36.36 0 0 1 0-.1l3.22-8.8a.2.2 0 0 1 .18-.14h12.54c.11 0 .19.11.18.33l-1.22 4.08H6.91a1.41 1.41 0 0 0-1.23 1l-1.41 3.9zm2.36 5.87a.16.16 0 0 1-.14-.07.27.27 0 0 1-.06-.17.3.3 0 0 1 0-.1L5 12.24l1.76-4.75a.2.2 0 0 1 .18-.14h12.54c.11 0 .19.11.18.33l-1.22 4.08L17 16.49a.2.2 0 0 1-.19.16H3.74zm16.42 5.72a.19.19 0 0 1-.18.16H6.88a.17.17 0 0 1-.14-.07.32.32 0 0 1 0-.18.28.28 0 0 1 0-.09l1.48-4.07h8.65a1.31 1.31 0 0 0 1.09-.66 2 2 0 0 0 .2-.45l1.14-3.79h3.36c.11 0 .2.11.19.33z"
                />
              </svg>
            </div>
            <div
              data-enzyme-id="Block-RenderedElement"
              style={
                Object {
                  "marginBottom": "8px",
                  "marginLeft": "0px",
                  "marginRight": "0px",
                  "marginTop": "0px",
                }
              }
            >
              <h3
                className="SectionTitle SectionTitle--uppercase"
              >
                 
                Data Types
                 
              </h3>
            </div>
            <p
              className="Text"
            >
              Deep dive into types, equality, coercion, immutability and more.
            </p>
          </Link>
          <Link
            className="TrackCard"
            to="/courses/Promises/"
          >
            <div
              className="TrackCard-iconWrap"
            >
              <svg
                className="Icon TrackCard-icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.1.49A12.88 12.88 0 0 0 1.31 5.08 1.006 1.006 0 0 0 3 6.17a10.94 10.94 0 0 1 11.6-3.78 10.06 10.06 0 0 1 7.4 9.26l-3-2.19a1 1 0 0 0-1.4.2 1 1 0 0 0 .2 1.39l4.5 3.29.6.2h.4a1 1 0 0 0 .6-.8A12.07 12.07 0 0 0 15.1.49zM21 17.72a11 11 0 0 1-11.6 3.79A10.06 10.06 0 0 1 2 12.25l3 2.19a1 1 0 1 0 1.2-1.6L1.71 9.56a1 1 0 0 0-1.4.19 1 1 0 0 0-.2.5A12.06 12.06 0 0 0 8.8 23.5a13.39 13.39 0 0 0 3.5.5 12.63 12.63 0 0 0 10.29-5.08 1 1 0 0 0-.2-1.39 1 1 0 0 0-1.39.19z"
                />
              </svg>
            </div>
            <div
              data-enzyme-id="Block-RenderedElement"
              style={
                Object {
                  "marginBottom": "8px",
                  "marginLeft": "0px",
                  "marginRight": "0px",
                  "marginTop": "0px",
                }
              }
            >
              <h3
                className="SectionTitle SectionTitle--uppercase"
              >
                 
                Promises
                 
              </h3>
            </div>
            <p
              className="Text"
            >
              Learn the Promise API inside and out.
            </p>
          </Link>
          <Link
            className="TrackCard"
            to="/courses/Testing/"
          >
            <div
              className="TrackCard-iconWrap"
            >
              <svg
                className="Icon TrackCard-icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16 2.75a9.39 9.39 0 1 0-1.29 14.34 2.05 2.05 0 0 0 .54 1l5.33 5.33a2 2 0 0 0 2.81-2.81l-5.33-5.33a2.05 2.05 0 0 0-1-.54A9.39 9.39 0 0 0 16 2.75zm-1.66 11.59a7 7 0 1 1 0-9.91 7 7 0 0 1 0 9.91z"
                />
              </svg>
            </div>
            <div
              data-enzyme-id="Block-RenderedElement"
              style={
                Object {
                  "marginBottom": "8px",
                  "marginLeft": "0px",
                  "marginRight": "0px",
                  "marginTop": "0px",
                }
              }
            >
              <h3
                className="SectionTitle SectionTitle--uppercase"
              >
                 
                Testing
                 
              </h3>
            </div>
            <p
              className="Text"
            >
              Learn frontend testing with Jest and Enzyme.
            </p>
          </Link>
        </div>
      </div>
      <div
        className="Landing-callOut"
      >
        <p>
          Awesome Learning is Frontend focused learning platform built around
           
          <b>
            deliberate practice
          </b>
          .
        </p>
        <p>
          We designed the courses to be perfect for 
          <b>
            group programming
          </b>
          .
        </p>
      </div>
      <div
        className="ContentSection"
      >
        <header
          className="ContentSection-title ContentSection-title--textLight"
        >
          <h1>
            What's Deliberate Practice?
          </h1>
        </header>
        <div
          className="ContentSection-content Landing-description ContentSection-content--alignLeft"
        >
          <p>
            Deliberate practice involves more than just repetition; it requires activities that are designed to improve performance, challenge the learner, and provide feedback.
          </p>
          <a
            className="Link Link--secondary"
            href="https://pwp.gatech.edu/bmeac/2016/02/18/deliberate-practice-and-why-you-should-do-it/"
            target="_blank"
          >
            Learn More
          </a>
        </div>
      </div>
      <div
        className="ContentSection"
      >
        <header
          className="ContentSection-title ContentSection-title--textLight"
        >
          <h1>
            How Do We Apply Deliberate Practice?
          </h1>
        </header>
        <div
          className="ContentSection-content Landing-description ContentSection-content--alignLeft"
        >
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            Each course and lesson are designed to incrementally challenge the learner while providing immediate feedback in the form of quizzes, provided solutions, and guiding comments.
          </p>
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            <b>
              Each course
            </b>
             is comprised of multiple lessons that each build on learnings from the previous lesson, always challenging the learner to reach for new understanding.
          </p>
          <p>
            <b>
              Each lesson
            </b>
             features exercises designed to incrementally increase in complexity to further challenge the user.
          </p>
        </div>
      </div>
      <div
        className="ContentSection"
      >
        <header
          className="ContentSection-title ContentSection-title--textLight"
        >
          <h1>
            Group Focused
          </h1>
        </header>
        <div
          className="ContentSection-content Landing-description ContentSection-content--alignLeft"
        >
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            We believe one of the major separators between "junior" and "senior" developers is technical communication. That's why we strongly suggest running these sessions with a small group, either in person or over a chat client.
          </p>
          <p>
            Over the hundreds of sessions we've run, those who tackle these courses as a group learn faster and become stronger technical communicators, all while building key technical skills.
          </p>
        </div>
      </div>
      <div
        className="ContentSection"
      >
        <header
          className="ContentSection-title ContentSection-title--textLight"
        >
          <h1>
            How it Works
          </h1>
        </header>
        <div
          className="ContentSection-content  ContentSection-content--alignLeft"
        >
          <ol
            className="Landing-list"
          >
            <li
              className="Landing-listItem"
            >
              <h3>
                Gather a Team
              </h3>
              <p
                className="Landing-listItemText"
              >
                We recommend groups of no more than
                 
                <a
                  className="Link Link--secondary"
                  href="https://en.wikipedia.org/wiki/Ringelmann_effect"
                  target="_blank"
                >
                  five or six people
                </a>
                . Try to build groups with relatively similar abilities across members to keep everyone interested and engaged.
              </p>
            </li>
            <li
              className="Landing-listItem"
            >
              <h3>
                Pick a Course
              </h3>
              <p
                className="Landing-listItemText"
              >
                Start with one of our pre-built courses like array methods, testing, etc. Later you can use the same learning method with other materials and contribute your own course!
              </p>
            </li>
            <li
              className="Landing-listItem"
            >
              <h3>
                Pick a Time
              </h3>
              <p
                className="Landing-listItemText"
              >
                We recommend setting aside an hour to an hour and a half per session, at a cadence of once per week until you complete your course.
              </p>
            </li>
            <li
              className="Landing-listItem"
            >
              <h3>
                Pick a Place
              </h3>
              <p
                className="Landing-listItemText"
              >
                Ideally, each of you has a laptop in a quiet space. Otherwise, find the best way to get your group together at the scheduled time whether in person or remotely.
              </p>
            </li>
          </ol>
        </div>
      </div>
      <div
        className="ContentSection"
      >
        <header
          className="ContentSection-title ContentSection-title--textLight"
        >
          <h1>
            Who is This For?
          </h1>
        </header>
        <div
          className="ContentSection-content Landing-description ContentSection-content--alignLeft"
        >
          <p
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            This material is currently used by full time software engineers within Wayfair. We believe beginners, experts, and everyone in between can learn something here.
          </p>
          <p>
            In general, we find that a group of folks motivated to learn and improve both their soft skills and Frontend knowledge will do well here. If there are explicit pre-requisites for a course, they will be clearly stated.
          </p>
        </div>
      </div>
      <div
        className="ContentSection"
      >
        <header
          className="ContentSection-title ContentSection-title--textLight"
        >
          <h1>
            Ready to Get Started?
          </h1>
        </header>
        <div
          className="ContentSection-content  ContentSection-content--alignLeft"
        >
          <div
            data-enzyme-id="Block-RenderedElement"
            style={
              Object {
                "marginBottom": "16px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
              }
            }
          >
            <Link
              activeClassName=""
              className="Link Link--secondary"
              to="/courses"
            >
              See Our Courses
            </Link>
          </div>
          <Link
            activeClassName=""
            className="Link Link--secondary"
            to="/howTo"
          >
            Read the Session How-To Guide
          </Link>
        </div>
      </div>
    </div>
  </div>,
]
`;
